chain-vote/src/gang/p256k1.rs

```diff
@@ -110,6 +118,11 @@ impl GroupElement {
         })
     }
 
+    pub(super) fn encode_hash_map(&self) -> Option<[u8; FieldElement::SIZE_BYTES]> {
```

Comment:
I'm still a bit skeptical about this. The name of this method suggest to me a behavior that should be implemented already by the `Hash` trait. 
The reason we were not using the `Hash` trait directly in bsgs is to do some optimizations, but those optimizations are only valid in that algorithm context and should be explicit (i.e. it's not something that the `GroupElement` can hide from bsgs, otherwise we won't be able to recover the point later).
I would rather keep the old approach of explicitly using only the x coordinate for sec2 curves